### DISCUSSION TITLE
How to get openVDB running in visual studio community 2022, for c++?

### DISCUSSION THREAD

**Discussion by Guneet-NF** [AcademySoftwareFoundation/openvdb#1744] [Q&A] [answered]
So i have followed the installation process for installing the library as mentioned in the git repo. 
But i am clueless as to what to do after that?
how to run a C++ openVDB code?
I am unable to find any header files to link in visual studios, and i can only find lib files.
Someone please help here

**Comment by mbytesdev** [comment]
The default install location should be `C:\Program Files\OpenVDB\`
Then the header files should be somewhere here: `C:\Program Files\OpenVDB\include\openvdb\openvdb.h` 
or for nanovdb `C:\Program Files\OpenVDB\include\nanovdb\...`
include it like `#include <C:\Program Files\OpenVDB\include\openvdb\openvdb.h>`

Alternative if you don't need to build it yourself, you can use the vcpkg version provided.
Pro: for the vcpkg version you can include the headers like `#include <openvdb/openvdb.h>`
Con: you are dependent on the versions as is (as far as I know)

I worked on a project for my studies and did a "How to get started" maybe it'll help you:
https://mbytesdev.github.io/vdb/

**Comment by cgolchert** [comment]
> So i have followed the installation process for installing the library as mentioned in the git repo. But i am clueless as to what to do after that? how to run a C++ openVDB code? I am unable to find any header files to link in visual studios, and i can only find lib files. Someone please help here

@Guneet-NF 

Did you do everything through vcpkg?  Did you also run `vcpkg integrate` as an admin? 

With those add:
_include directories_ to (vcpkg location)\installed\x64-windows\include
_library directories_ to (vcpkg location)\installed\x64-windows\bin and (vcpkg location)\installed\x64-windows\lib

for example mine are:
c:\dev\vcpkg\installed\x64-windows\include

c:\dev\vcpkg\installed\x64-windows\bin
c:\dev\vcpkg\installed\x64-windows\lib

That should let you use `#include <openvdb/openvdb.h>` and you should be off running.

**Reply by Guneet-NF** [reply]
So after installing vcpkg, and following everything according to the installation steps given on openvdb git repo for windows, 
I found the vdb files in C:\Program Files\OpenVDB
I could not find openvdb related files in vcpkg\installed\x64-windows
how did you do everything using vcpkg tho?

**Reply by cgolchert** [reply]
I just looked at the steps they give.  I'm not sure why they have to grab it from their git instead of letting vcpkg take care of it unless this is to get around any delay in vcpkg porting the latest version.  This is just using vcpkg for the dependencies.*

I'm GUESSING you can just drill down into the C:\Program Files\OpenVDB as your library and include directories.  



*I'll have to check that out to see if it avoids a dll collision I'm getting with a 3d app.

**Reply by cgolchert** [reply, marked as answer]
To do everything in vcpkg just tell it `vcpkg install openvdb`  That will build all the dependencies AND the current port of openvdb (11.0 at the time of this writing)

**Reply by Gnomusy** [reply]
Hi, cgolchert, I did that, but when I run a simple cookbook code:

No problem with this line:
// Initialize the OpenVDB library
openvdb::initialize();

But with this line:
// Create an empty floating-point grid with background value 0.
    openvdb::FloatGrid::Ptr grid = openvdb::FloatGrid::create();

I get the error: LNK2001 external symbol imath_half_to_float_table unresolved

What I did wrong?
What can I do to solve this?
Any help wil be really welcome!

David - Gnomusy

**Reply by Guneet-NF** [reply]
Write "#define IMATH_HALF_NO_LOOKUP_TABLE" before you include openvdb
then it should work

**Reply by Gnomusy** [reply]
A thousand thanks, Gunnet-NF, this is paramount for my project.
It works perfectly!
Regards
David
